### DISCUSSION TITLE
Easy access to the route object from script?

### DISCUSSION THREAD

**Discussion by AndersNielsen85** [nuxt/nuxt#18330] [Questions] [answered]
You can access the route object with {{ $route }} in <template>

In composition-api (https://github.com/nuxt-community/composition-api) package a useRoute wrapper is available, but I can't seem to find anything similar directly in nuxt3.

It is available trough _legacyContext, but it would be nice for some easier access, not digging into something named legacy :)?
```
const nuxtApp = useNuxtApp();
console.log(nuxtApp._legacyContext.route);
```

**Comment by AndersNielsen85** [comment, marked as answer]
Occured to me that it can be accessed directly from vue-router with vue3 :)

```
import { useRoute } from 'vue-router';

const route = useRoute()
console.log(route);
```

**Reply by danielroe** [reply]
That is exactly how to do it. We event auto-import it for you from `vue-router` so you don't have to import it yourself 🙃
